### ISSUE TITLE
Add images to documents

### ISSUE THREAD

**Issue by nico202** [styx-static/styx#37]
Hi, to me it's not clear how to add static files.

How do I tell styx to copy the whole content of a folder in the derivation?

Thanks, Nicolò

**Comment by ericsagnes** [comment]
You can pass a list of files to copy to the resulting derivation to the `files` parameter of `mkSite`. 
https://styx-static.github.io/styx-site/documentation/library.html#lib.generation.mkSite

The recommended approach is to make a custom [theme](https://styx-static.github.io/styx-site/documentation/index.html#_creating_a_new_theme) and put the files in its `files` folder.

**Comment by nico202** [comment]
Ehm.. This might be a problem, since it seems to me that's not possible to include images generated _during_ the doc->html conversion, right?

**Comment by ericsagnes** [comment]
> Ehm.. This might be a problem, since it seems to me that's not possible to include images generated during the doc->html conversion, right?

Right, it is not possible. The doc -> html conversion is done with `markupToHtml` that just convert a markup text into html.
 
I cannot think of images generated from asciidoc or markdown, is there an orgmode use case for this?

**Comment by nico202** [comment]
Yes, org is able to generate images like [this](https://kotfic.github.io/org-mode-export-of-matplotlib-images-etc.html).
My aim was to have a "literate programming" blog fully reproducible, and generating images would be extremely important

**Comment by nico202** [comment]
An alternative could be to embed images as base64 like [here](https://emacs.stackexchange.com/questions/27060/embed-image-as-base64-on-html-export-from-orgmode). But still this would work for images only and it would be impossible to cross-reference images between different pages.

**Comment by ericsagnes** [comment]
Or in embedded SVG, but that's not a very satisfying too.
The could be something to do with `data` and embedded nix.

**Comment by nico202** [comment]
Yeh embedding things is not the best choice.. Also, not only images should be supported since org-mode is able to "tangle" content to external files, plus sometimes it might be useful to generate arbitrary binary data. If this is off the aim of styx I might go in some other way though
